flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/UnionInputGate.java

```diff
@@ -307,4 +309,18 @@ public void registerBufferReceivedListener(BufferReceivedListener listener) {
 			inputGate.registerBufferReceivedListener(listener);
 		}
 	}
+
+	@Override
+	public Collection<Buffer> getInflightingBuffers(int channelIndex, long checkpointId) throws IOException {
+		int totalChannels = 0;
+		for (InputGate inputGate : inputGates) {
+			int numberOfChannels = inputGate.getNumberOfInputChannels();
+			if (numberOfChannels + totalChannels > channelIndex) {
+				inputGate.getInflightingBuffers(channelIndex - totalChannels, checkpointId);
+				break;
+			}
+			totalChannels += numberOfChannels;
+		}
+		throw new IllegalArgumentException("Invalid channel index");
```

Comment:
This method is missing some valid `return`?

Reply:
Yes, it should add return in above `return inputGate.getInflightingBuffers(channelIndex - totalChannels, checkpointId);`